src/tmiic.cpp

```diff
@@ -0,0 +1,173 @@
+//
+// Detail: For the orientation step when latent variable discovery is enabled, 
+// we duplicate the edges over the history, assuming stationarity, to improve 
+// the orientation step
+//
+// Params: 
+// - Environment&: the environment structure
+//
+// Returns: none
+//-----------------------------------------------------------------------------
+void repeatEdgesOverHistory (Environment& environment) {
+  //
+  // Now, we iterate over computed edges to duplicate (if needed)
+  // the edges over the history
+  //
+  auto& edge_list = environment.connected_list;
+  auto edge_end = end(edge_list);
+  for (auto iter0 = begin(edge_list); iter0 != edge_end; ++iter0) {
+    //
+    // Find the Edge structure that will be duplicated 
+    //
+    int node1_pos = iter0->X;
+    int node2_pos = iter0->Y;
+    const Edge& edge_orig = environment.edges(node1_pos, node2_pos);
+    //
+    // To create the same edge over history, we switch the nodes pos by n_nodes_not_lagged
+    // until node pos > nb total nodes
+    //
+    while (true) {
+      node1_pos += environment.n_nodes_not_lagged;
+      node2_pos += environment.n_nodes_not_lagged;
+      if ( (node1_pos >= environment.n_nodes) || (node2_pos >= environment.n_nodes) )
+        break;
+     //
+      // Duplicate the edge info into environment.edges array
+      //
+      Edge& edge_to_modif = environment.edges(node1_pos,node2_pos);
+      edge_to_modif.status = edge_orig.status;
+      edge_to_modif.status_init = edge_orig.status_init;
+      edge_to_modif.status_prev = edge_orig.status_prev;
+      //
+      // Add the nodes into connected_list
+      //
+      environment.connected_list.emplace_back (node1_pos, node2_pos, environment.edges(node1_pos,node2_pos) );
+
+      Edge& edge_to_modif_inverse = environment.edges(node2_pos,node1_pos);
+      edge_to_modif_inverse.status = edge_orig.status;
+      edge_to_modif_inverse.status_init = edge_orig.status_init;
+      edge_to_modif_inverse.status_prev = edge_orig.status_prev;
+    }
+  }
+}
+
+//-----------------------------------------------------------------------------
+// dropPastEdges
+//-----------------------------------------------------------------------------
+// Description: 
+// Drop past edges (the edges having no node of the final timestep)
+//
+// Detail: In orientation step, when latent variable discovery is enabled, 
+// we duplicated stationary edges over history. Here, we ensure that the edges
+// are restored to their state prior to orientation. 
+//
+// Params: 
+// - Environment&: the environment structure
+//
+// Returns: none
+//--------------------------------------------------------------------------------
+void dropPastEdges (Environment& environment) {
+  //
+  // We iterate over computed edges to find edges previously duplicated using stationary.
+  // All the edges duplicated are disconnected and removed from environment.connected_list
+  //
+  std::regex lag_expr(".*lag");
```

Comment:
As discussed earlier, it's highly unrecommended to use string and regex here, since all of the "lagged" framework is constructed by tmiic, it's better to use a index-based rule to control the lagged nodes, same apply for the related part in R

Reply:
As discussed effectively, since all of the "lagged" framework is constructed by tmiic as you mention, we are sure the nodes names received in the C++ are lagged so IMHO, it is not an issue here to test on string part because these string parts are added by our program. I can look to not use regex but I would prefer to keep tests on strings as the alternatives would be:
- add an extra parameter in the R/C++ interface but as I already mentioned,  IMO the interface should be kept minimal.
- lag the input data in the C++ part so the C++ part has all the information but we needs to carefully check what does miic.R and miic.reconstruct.R as all the code in these two files will operate on non lagged data. It can be solved by moving also the code of miic.R and reconstruct.R into the C++ but it is probably a bunch of work

Reply:
- See the discussion in the previous hunk about the interface. About the regex test on string, you are right that it's not an issue, but we are not talking about a bug here. The idea is that what you want to do can be done with simple operation on the indices, why make things more complicated?

- I don't quite get your idea here. Technically there's no restriction on what should be done in C++ and what should be done in R. The rule we are following in miic is to have C++ done the computation, while leaving pre(post) processing to R. What you want to do with these C++ functions belong to pre processing of the lagged data, and can be easily achieved in R.